### PR TITLE
Fix problems with extern block improvement PR.

### PR DESCRIPTION
**Pull request by mppf**
Follow-up to PR #21944

PR #21944 improved extern block in the dyno scope resolver by adding integration with clang. However, there were some problems with computing the correct clang arguments, especially when `CHPL_LLVM_GCC_PREFIX` is used. Additionally, there were problems with LLVM 11 and 12 since SHA256 was added to the LLVM Support library in LLVM 13. This PR fixes these problems.-

Reviewed by @riftEmber - thanks!

- [x] full local testing

### CODE REVIEW

File: frontend/lib/util/clang-integration.cpp [hunk 1]
```diff
@@ -54,6 +54,7 @@ namespace util {
 const std::vector<std::string>& clangFlags(Context* context) {
   QUERY_BEGIN_INPUT(clangFlags, context);
   std::vector<std::string> ret;
+  ret.push_back("clang"); // dummy argv[0]
```
Comment (riftEmber) [inline frontend/lib/util/clang-integration.cpp:57]: Why is this necessary -- are we ever calling `clangFlags` and assuming there is something in it before it's been set? I see we `setClangFlags` from the output of `computeClangArgs` which with this PR is guaranteed to have the `clang` executable name as the 0th element already.

File: frontend/lib/util/clang-integration.cpp [hunk 2]
```diff
@@ -143,15 +129,17 @@ const std::vector<std::string>& getCC1Arguments(Context* context,
...
+  clang::driver::Driver D(argsCstrs[0], triple, *diags);
 
   std::unique_ptr<clang::driver::Compilation> C(D.BuildCompilation(argsCstrs));
 
   clang::driver::Command* job = nullptr;
 
   if (usingGpuLocaleModel(context) == false) {
-    // Not a CPU+GPU compilation, so just use first job.
-    job = &*C->getJobs().begin();
+    if (C->getJobs().begin() != C->getJobs().end()) {
```
Comment (riftEmber) [inline frontend/lib/util/clang-integration.cpp:139]: nitpick, but it looks like this is just to check there are jobs -- it would be more readable to just directly use `!C->getJobs().empty()` ([docs](https://clang.llvm.org/doxygen/classclang_1_1driver_1_1JobList.html))

**Review by riftEmber — APPROVED** [review]

File: frontend/lib/util/clang-integration.cpp (hunk 1 above)
Comment (mppf) [reply frontend/lib/util/clang-integration.cpp:57]: Yes, we call clangFlags without ever calling setClangFlags in some of the dyno C++ tests. So, it is useful to have something plausible for that case. I will add a comment that it's only expected to be used this way in tests.
